atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -621,7 +622,17 @@ private <T> TransactionKeyValueServiceManager createTransactionKeyValueServiceMa
                         metricsManager,
                         config().initializeAsync());
         return factory.create(
-                coordinationService, keyValueServiceManager, config, runtimeConfigRefreshable, initializeAsync);
+                config().namespace()
+                        .orElseThrow(() -> new SafeIllegalArgumentException(
+                                "Namespace must be set when using transactionKeyValueService")),
+                reloadingFactory()
```

Comment:
@jeremyk-91 we should clean this up, there's quite a few places where UserAgent is passed explicitly, where you could really just hide it inside the ReloadingFactory.

Reply:
yeah this is probably from the pre-dialogue move

Reply:
Yeah, I suspect this is remoting -> conjure.